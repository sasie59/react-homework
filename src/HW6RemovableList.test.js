import React from 'react';
import { getByTestId, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW from './HW6RemovableList';

test('should HW6RemovableList', () => {
  const { container } = render(<HW />);
  expect(container).toMatchSnapshot();
});

test('should add || remove something to list', () => {
  const { container,getByTestId } = render(<HW />);
  const input = container.querySelector('input');
  
  userEvent.type(input, "remove{enter}");
  userEvent.click(getByTestId(0));

  expect(container).toMatchSnapshot();
});
test('should add || remove something to list', () => {
  const { container,getByTestId } = render(<HW />);
  const input = container.querySelector('input');
  
  userEvent.type(input, "remove{enter}");
  userEvent.type(input, "test{enter}");

  userEvent.click(getByTestId(1));

  expect(container).toMatchSnapshot();
});
