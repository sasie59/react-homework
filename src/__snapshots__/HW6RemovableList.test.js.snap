// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should HW6RemovableList 1`] = `
<div>
  <div>
    <h1>
      HW6RemovableList
    </h1>
    <form>
      <input
        type="text"
      />
    </form>
    <ul />
  </div>
</div>
`;

exports[`should add || remove something to list 1`] = `
<div>
  <div>
    <h1>
      HW6RemovableList
    </h1>
    <form>
      <input
        type="text"
      />
    </form>
    <ul />
  </div>
</div>
`;

exports[`should add || remove something to list 2`] = `
<div>
  <div>
    <h1>
      HW6RemovableList
    </h1>
    <form>
      <input
        type="text"
      />
    </form>
    <ul>
      <li>
        test
        <button
          data-testid="0"
        >
          Clear it!
        </button>
      </li>
    </ul>
  </div>
</div>
`;
